### PR TITLE
[REF] spreadsheet: convert X2manyTagSelector to wowl components

### PR DESCRIPTION
**Pull request by LucasLefevre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/global_filters/components/records_selector.js [hunk 1]
```diff
@@ -93,17 +91,17 @@ X2ManyTagSelector.template = xml/*xml*/ `
...
      * (selectedRecords: Array<{ id: number; display_name: string }>) => void
      **/
     onValueChanged: Function,
-    relatedModel: String,
+    resModel: String,
     /**
      * Array of selected record ids
      */
-    selectedValues: {
+    resIds: {
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/src/global_filters/components/records_selector.js]: other idea ? any thoughts ? `activeIds` ? `selectedIds` ?

File: addons/spreadsheet/static/src/global_filters/components/tag_selector_widget.js [hunk 2]
```diff
@@ -1,174 +1,108 @@
...
+X2ManyTagSelector.components = { TagsList, Many2XAutocomplete };
 X2ManyTagSelector.template = xml/*xml*/ `
-    <TagSelectorWidgetAdapter
-        Component="TagSelectorWidget"
-        relatedModel="props.relatedModel"
-        selectedValues="props.selectedValues"
-        onValueChanged="props.onValueChanged"
-        placeholder="props.placeholder"
-    />`;
+    <div class="o_field_widget o_field_many2many_tags">
```
Comment (pro-odoo) [inline addons/spreadsheet/static/src/global_filters/components/tag_selector_widget.js]: Template should be defined in an xml file

File: addons/spreadsheet/static/src/global_filters/components/tag_selector_widget.js [hunk 3]
```diff
@@ -1,174 +1,108 @@
...
+                    getDomain.bind="searchDomain"
+                    isToMany="true"
+                />
+            </div>
+        </div>
+    </div>`;
 X2ManyTagSelector.props = {
+    /**
+     * Callback called when a record is selected or removed.
+     * (selectedRecords: Array<{ id: number; display_name: string }>) => void
```
Comment (pro-odoo) [inline addons/spreadsheet/static/src/global_filters/components/tag_selector_widget.js]: 👍

File: addons/spreadsheet/static/src/global_filters/components/records_selector.js (hunk 1 above)
Comment (pro-odoo) [reply addons/spreadsheet/static/src/global_filters/components/records_selector.js]: resIds is perfect IMO

**Review by pro-odoo — APPROVED** [review]

File: addons/spreadsheet/static/src/global_filters/components/tag_selector_widget.js (hunk 2 above)
Comment (pro-odoo) [reply addons/spreadsheet/static/src/global_filters/components/tag_selector_widget.js]: Did in the 3rd commit --'

**Comment by pro-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**